pkg/apis/core/validation/validation.go

```diff
@@ -1321,6 +1321,32 @@ func validateMountPropagation(mountPropagation *core.MountPropagationMode, conta
 	return allErrs
 }
 
+// ValidateMountRecursiveReadOnly validates RecursiveReadOnly mounts.
+func ValidateMountRecursiveReadOnly(mount core.VolumeMount, fldPath *field.Path) field.ErrorList {
+	if mount.RecursiveReadOnly == nil {
+		return nil
+	}
+	allErrs := field.ErrorList{}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "recursive read-only mount mode needs feature gate RecursiveReadOnlyMounts"))
+	}
+	switch *mount.RecursiveReadOnly {
+	case core.RecursiveReadOnlyDisabled:
+		// NOP
+	case core.RecursiveReadOnlyEnabled, core.RecursiveReadOnlyIfPossible:
+		if !mount.ReadOnly {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "recursive read-only mount mode is available only to read-only mounts"))
+		}
+		if mount.MountPropagation != nil && *mount.MountPropagation != core.MountPropagationNone {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "recursive read-only mount mode require propagation to be None or unset"))
```

Comment:
"may only be specified when mountPropagation is None or not specified"

Reply:
fixed